Fix CodeQL config error

https://github.com/scabench/l4j-tp1/actions/runs/7453693386/job/20279601718 shows a failing GitHub CodeQL run using Autobuild. It doesn't like JDK17:

```
Error: 1-08 22:02:54] [autobuild] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project l4j-tp1: Fatal error compiling: error: invalid target release: 17 -> [Help 1]
```

It's confusing that Autobuild is running at all, since the latest commit https://github.com/scabench/l4j-tp1/commit/a4e07f918696616bb77a18ab13acb8ea5728843a happened before this run, and that commit overrides Autobuild with a manual `mvn compile` command.

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -60,6 +60,11 @@ jobs:
         # queries: security-extended,security-and-quality
 
 
+    # As "documented" at https://github.com/github/codeql-action/issues/825#issuecomment-979311250
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 17
+
     # Autobuild attempts to build any compiled languages (C/C++, C#, Go, Java, or Swift).
     # If this step fails, then you should remove it and run the build manually (see below)
     # - name: Autobuild

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -72,8 +72,8 @@ jobs:
     #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
 
     - run: |
-      echo "Run, Build Application using script"
-      mvn compile
+        echo "Run, Build Application using script"
+        mvn compile
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v3

